File: src/handlers/error/error-handler-factory.ts
import { Converter } from "../../core/converter";
import { MethodMode } from "../../types";
import { BaseErrorHandler } from "./base-error";
import { EventErrorHandler } from "./event-error";
import { ServiceErrorHandler } from "./service-error";

export class ErrorHandlerFactory {
    public static getHandler(converter: Converter) {
        switch (converter.getMethodMode()) {
            case MethodMode.EVENT:
                return new EventErrorHandler(converter);
            case MethodMode.SERVICE:
                return new ServiceErrorHandler(converter);
            default:
                return new BaseErrorHandler(converter);
        }
    }
}

File: src/main/java/co/uk/stefanpuia/minilang2java/core/handler/error/ErrorHandlerFactory.java
package co.uk.stefanpuia.minilang2java.core.handler.error;

import co.uk.stefanpuia.minilang2java.core.convert.context.ConversionContext;
import lombok.experimental.UtilityClass;

@UtilityClass
public class ErrorHandlerFactory {
  public static ErrorHandler newInstance(final ConversionContext context) {
    return switch (context.getMethodMode()) {
      case SERVICE -> new ServiceErrorHandler(context);
      case EVENT -> new EventErrorHandler(context);
      default -> new UtilErrorHandler(context);
    };
  }
}
